internal/process.go

```diff
@@ -560,3 +624,64 @@ func consumeReports(client rest.Client, exporter export2.Exporter, workerID int,
 		}
 	}
 }
+
+func getPresetFileName(fileName string) string {
+	return path.Join(export2.PresetsDirName, fileName)
+}
+
+func producePresets(presetList []*rest.PresetShort, presetJobs chan<- PresetJob) {
+	for _, v := range presetList {
+		presetJobs <- PresetJob{
+			PresetID: v.ID,
+		}
+	}
+	close(presetJobs)
+}
+
+func consumePresets(soapClient interfaces.PresetProvider, exporter export2.Exporter, workerID int,
+	presetJobs <-chan PresetJob, done chan<- PresetConsumeOutput) {
+	for presetJob := range presetJobs {
+		l := log.With().
+			Int("PresetID", presetJob.PresetID).
+			Int("worker", workerID).
+			Logger()
+		presetData, presetErr := getPresetData(soapClient, presetJob.PresetID, workerID)
+		if presetErr != nil {
+			l.Debug().Err(presetErr).Msgf("failed creating preset after %d attempts", presetCreateAttempts)
+			done <- PresetConsumeOutput{Err: presetErr, PresetID: presetJob.PresetID}
+			continue
+		}
+
+		if errExp := exporter.AddFile(getPresetFileName(fmt.Sprintf("%d.xml", presetJob.PresetID)), presetData); errExp != nil {
+			err := errors.Wrapf(errExp, "error with exporting preset %d list to file", presetJob.PresetID)
+			done <- PresetConsumeOutput{Err: err, PresetID: presetJob.PresetID}
+			continue
+		}
+
+		done <- PresetConsumeOutput{Err: nil, PresetID: presetJob.PresetID}
+	}
+}
+
+func getPresetData(soapClient interfaces.PresetProvider, presetID, workerID int) ([]byte, error) {
```

Comment:
This resilience code seems to belong in the soap client.
Please consider using a RetryableHTTPAdapter in soap client, like the rest client does. That would give resilience to all requests and this function would not be needed.